modules/gateway-task/main.tf

```
@@ -70,135 +84,256 @@ resource "aws_ecs_task_definition" "this" {
-          // AWS API will accept invalid values, and you will see a CannotStartContainerError at runtime.
-          softLimit = 1048576
-          hardLimit = 1048576
-        }]
-      },
-    ],
+        }] : [],
+      )
+    )
   )
 }
+
+resource "aws_ecs_service" "this" {
+  name            = local.service_name
+  cluster         = var.ecs_cluster_arn
+  task_definition = aws_ecs_task_definition.this.arn
+  desired_count   = 1
+  network_configuration {
+    subnets          = var.subnets
+    security_groups  = local.security_groups
+    assign_public_ip = var.assign_public_ip
+  }
+  dynamic "load_balancer" {
+    for_each = local.load_balancer
+    content {
+      target_group_arn = load_balancer.value["target_group_arn"]
+      container_name   = load_balancer.value["container_name"]
+      container_port   = load_balancer.value["container_port"]
+    }
+  }
+  launch_type            = var.launch_type
+  propagate_tags         = "TASK_DEFINITION"
+  enable_execute_command = true
+}
+
+resource "aws_lb" "this" {
+  count              = var.lb_enabled ? 1 : 0
+  name               = local.service_name
+  internal           = false
+  load_balancer_type = "network"
+  subnets            = var.lb_subnets
+}
+
+resource "aws_lb_target_group" "this" {
+  count                = var.lb_enabled ? 1 : 0
+  name                 = local.service_name
+  port                 = tostring(local.wan_port)
+  protocol             = "TCP"
+  target_type          = "ip"
+  vpc_id               = var.lb_vpc_id
+  deregistration_delay = 120
+  health_check {
+    protocol = "TCP"
+  }
+}
+
+resource "aws_lb_listener" "this" {
+  count             = var.lb_enabled ? 1 : 0
+  load_balancer_arn = aws_lb.this[0].arn
+  port              = tostring(local.wan_port)
+  protocol          = "TCP"
+
+  default_action {
+    type             = "forward"
+    target_group_arn = aws_lb_target_group.this[0].arn
+  }
+}
+
+resource "aws_security_group" "this" {
+  count       = var.lb_enabled && var.lb_create_security_group ? 1 : 0
+  name        = local.service_name
+  description = "Security group for ${local.service_name}"
+  vpc_id      = var.lb_vpc_id
+}
+
+// The Terraform module automatically removes the allow all egress rule when creating a security group.
+// When creating a security group we need to include the allow all egress rule.
+resource "aws_security_group_rule" "lb_egress_rule" {
+  count             = var.lb_enabled && var.lb_create_security_group ? 1 : 0
+  type              = "egress"
+  description       = "Egress rule for ${local.service_name}"
+  from_port         = 0
+  to_port           = 0
+  protocol          = "-1"
+  cidr_blocks       = ["0.0.0.0/0"]
+  ipv6_cidr_blocks  = ["::/0"]
+  security_group_id = aws_security_group.this[0].id
+}
+
+resource "aws_security_group_rule" "lb_ingress_rule" {
+  count             = (var.lb_enabled && var.lb_create_security_group) || var.lb_modify_security_group ? 1 : 0
```

nit: So that if we set lb_enabled = false, then this security group rule would not get created in any case:

```suggestion
  count             = var.lb_enabled && (var.lb_create_security_group) || var.lb_modify_security_group) ? 1 : 0
```